internal/cmd/api-key/command_delete.go

```diff
@@ -28,17 +25,42 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	if err := c.confirmDeletion(cmd, args); err != nil {
 		return err
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
+
+	deleted, err := resource.Delete(args, func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
+		}
+		return nil
+	}, c.postProcess)
```

Comment:
Why do we need to pass this function in? Can't we just call it after `resource.Delete()`?

Reply:
This is to avoid looping through all the id's twice, once to delete and once to perform additional tasks.